### DISCUSSION TITLE
Welcome to userscripts Discussions!

### DISCUSSION THREAD

**Discussion by quoid** [quoid/userscripts#128] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by svenkevs** [comment]
Hi all,

Great work on the plugin!
I had a small feature request: can it be possible to add a simple way to create a load-order for the scripts? Some scripts depend on another script being loaded first. If they are nicely made with an indicator for starting at document-start or document-end it usually is not an issue, but I noticed that scripts not specifying where to load it can be loaded too soon at times.

Ideally the plug-in would first load scripts that are to be loaded at the document start, then scripts that do not specify a position, then load the page, and then load scripts that specify loading at document end. Also a simple system for users to make sure some scripts are loaded first would really help already!

Nothing urgent, the plug-in allows us to easily edit scripts to tell scripts which don't specify it to '// @run-at      document-end'. Just a nice to have in the future :)

**Reply by quoid** [reply]
Thanks from dropping in @svenkevs 

Currently there's a unique metadata key called [`@weight`](https://github.com/quoid/userscripts#metadata) - does this not accomplish what you're looking to do or behaving buggy?

The link above should take you to the readme, `@weight` is documented in the `metadata` section

**Reply by svenkevs** [reply]
Thank you! It seems to accomplish it also.
So far I was fixing it by forcing other scripts to run only after a page is loaded, but now I can just put maximum weight to the one that has to be loaded first (less work as I need to maintain it in only one script after updates).

Do you happen to know what the default weight is for scripts that do not include weight metadata? 
I'll see if I can have that script's source repository updated to include a high weight by default :)

**Reply by quoid** [reply]
The default weight is `1` and the values accepted are [`1` to `999`](https://github.com/quoid/userscripts/blob/71094303ef8b10cd7861ccb1166c678d6b81ebf3/extension/Userscripts%20Extension/Functions.swift#L38).

I hope it works out for you
